controllers/datadogagent/feature/admissioncontroller/feature.go

```diff
@@ -83,6 +102,55 @@ func (f *admissionControllerFeature) Configure(dda *v2alpha1.DatadogAgent) (reqC
 		if ac.WebhookName != nil {
 			f.webhookName = *ac.WebhookName
 		}
+
+		sidecarConfig := dda.Spec.Features.AdmissionController.AgentSidecarInjection
+		if shouldEnablesidecarInjection(sidecarConfig) {
+			f.agentsidecarConfig = &AgentSidecarInjectionConfig{}
+			componentOverride, ok := dda.Spec.Override[v2alpha1.NodeAgentComponentName]
+			if sidecarConfig.Enabled != nil {
+				f.agentsidecarConfig.enabled = *sidecarConfig.Enabled
+
+			}
+			if sidecarConfig.Provider != nil && *sidecarConfig.Provider != "" {
+				f.agentsidecarConfig.provider = *sidecarConfig.Provider
+			}
+
+			if sidecarConfig.ClusterAgentCommunicationEnabled != nil {
+				f.agentsidecarConfig.clusterAgentCommunicationEnabled = *sidecarConfig.ClusterAgentCommunicationEnabled
+			} else {
+				f.agentsidecarConfig.clusterAgentCommunicationEnabled = apicommon.DefaultAdmissionControllerAgentSidecarClusterAgentEnabled
+			}
+
+			// set image registry from admissionController config or global config if defined
```

Comment:
where do you set it from global config?

Reply:
Maybe it would make these conditions easier if you set the defaults somewhere here and here you only handle Override and user provided `global.registry`
https://github.com/DataDog/datadog-operator/blob/f0e5f6be4421603604481d94e9fd3c30b37723c2/apis/datadoghq/v2alpha1/datadogagent_default.go#L128C20-L128C28